### PR TITLE
[Feature] Update button and link components

### LINKED ISSUE #6733: ⚙️ Update the button and link components
image

## 🌟 Details ([Figma file](https://www.figma.com/file/T1R64oNWog10OXiBnuiMkz/2.-Components?type=design&node-id=756-5888))

Updates to the button are minimal and mostly focus on sizing, but please review the information below to ensure both the button and link components (when styled as a button) are up-to-date.

### Accessibility

Buttons should allow for an optional property that can specify alternative, visually hidden AT text to allow for more descriptive button actions in instances where buttons are repeated visually and have enough context to be concise.

### Typography

Buttons should use a bolded, underlined, `copy` font size.

#### Block-style buttons

Note that block-style buttons should use the `rounded` corner radius.

image 

#### Inline-style buttons

image

#### Call-to-action buttons

Note that call-to-action buttons should use the `rounded` corner radius. Also note that the stroke width of the icon on call-to-action buttons should be increased to `2.5px` where possible.

image 

#### Block-style buttons

These are your standard "button" look and feel. They can either be sized based on their label, or span the available space, depending on the design context.

image

#### Inline-style buttons

These are your standard "link" look and feel, but retain the functionality of the button component (e.g. icons, etc.). This style doesn't contain any exterior padding to help them integrate seamlessly with other inline elements.

image

#### Call-to-action buttons

These are custom buttons used to direct attention to particular features, tasks, or call outs.

image

### Color variants

Note that the `IAP` theme only has `primary` and `secondary` branded button colors. All other branded palette options should default to the IAP's `secondary` color scheme, with the exception of stoplight colors and `black`, which are global across themes. IAP colors are described where applicable.

#### Primary

- For the `GCDT` theme
  - Uses the `primary.light` color for block-style backgrounds and `black` text/icons
  - Uses the `primary.darker` color for inline-style text/icons
  - Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `primary.light` for the icon background, and has a `medium` shadow applied to the whole element
- For the `IAP` theme
  - Uses the `primary.light` color for block-style backgrounds and `black` text/icons
  - Uses the `primary.darker` color for inline-style text/icons
  - Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `primary.light` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Secondary

- For the `GCDT` theme
  - Uses the `secondary` color for block-style backgrounds and `black` text/icons
  - Uses the `secondary.darker` color for inline-style text/icons
  - Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `secondary` for the icon background, and has a `medium` shadow applied to the whole element
- For the `IAP` theme
  - Uses the `secondary` color for block-style backgrounds and `white` text/icons
  - Uses the `secondary.darker` color for inline-style text/icons
  - Uses `foreground` for the white portion of the call-to-action, `black` text, a `white` icon, `secondary` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Tertiary

- Uses the `tertiary` color for block-style backgrounds and `black` text/icons
- Uses the `tertiary.darker` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `tertiary` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Quaternary

- Uses the `quaternary` color for block-style backgrounds and `black` text/icons
- Uses the `quaternary.darker` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `quaternary` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Quinary

- Uses the `quinary` color for block-style backgrounds and `black` text/icons
- Uses the `quinary.darker` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `quinary` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Error/destructive

- Uses the `error.light` color for block-style backgrounds and `black` text/icons
- Uses the `error.darker` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `error.light` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Warning

- Uses the `warning` color for block-style backgrounds and `black` text/icons
- Uses the `warning.darker` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `warning` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Success

- Uses the `success.light` color for block-style backgrounds and `black` text/icons
- Uses the `success.darker` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text/icons, `success.light` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Black

- Uses the `black` color for block-style backgrounds and `white` text/icons
- Uses the `black` color for inline-style text/icons
- Uses `foreground` for the white portion of the call-to-action, `black` text, a `white` icon, `black` for the icon background, and has a `medium` shadow applied to the whole element

image

#### Block-style buttons

- On hover, the button's background color should change to the relevant `lightest` modifier of the color used, with text remaining or switching to `black`
- On focus, the button's background color should change to the `focus` color, with text remaining or switching to `black`

image

#### Inline-style buttons

- On hover, the text/icon color should shift one shade up to the `dark` modifier
- On focus, the text/icon should change to `black` and the background should change to `focus`

image

#### Call-to-action buttons

- On hover, the button's shadow should increase to `larger` and the icon's background should change to the relevant `lightest` modifier for the color used
- On focus, the icon's background should change to `focus`, with the rest of the button remaining the same

image

### Block and inline buttons with icons

Buttons shouldn't have icons by default, but when they do have them, they should be set to `x.75` width/height, and be selected from Heroicon's `mini` collection.

image

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #6733 

## 👋 Introduction

Updated both the `Button` and `Link` components with new styles.

## 🕵️ Details

We had a some drifting styles of different links and buttons, this also attempts to merge those to keep things in sync.

The following have been removed and made props of the main components:
 - `ExternalLink`
 - `IconLink`
 - `IconButton`
 - `CallToActionLink`
 - `CallToActionLink`

## 📋 TO DO

- [x] Remove support for `outline`
- [x] Update small radii to `5px`
- [x] Adjust call to action hover
- [x] Improve disabled styles

#### Shared Types

```tsx
// Alllows use of `data-h2-*` on both components
type HydrogenAttributeKey = `data-h2-${string}`;
type HydrogenAttributes = {
  [dataAttribute: HydrogenAttributeKey]: unknown;
};

// Type used for passing in a heroicon as  a prop
type IconProps = React.PropsWithoutRef<React.SVGProps<SVGSVGElement>> & {
  title?: string;
  titleId?: string;
} & React.RefAttributes<SVGSVGElement>;

type IconType = React.ForwardRefExoticComponent<IconProps>;

// Available colour palette
type Color =
  | "primary"
  | "secondary"
  | "tertiary"
  | "quaternary"
  | "quinary"
  | "white"
  | "black"
  | "success"
  | "warning"
  | "error";

// Modes available
type ButtonLinkMode = "solid" | "outline" | "text" | "cta";

// Common props for both Button and Link's
type ButtonLinkProps = {
  color?: Color;
  block?: boolean;
  mode?: ButtonLinkMode;
  icon?: IconType;
} & HydrogenAttributes;
```

#### `<Link />` Props

```tsx
type LinkProps = ButtonLinkProps &
  Omit<RouterLinkProps, "to"> &
  Omit<
    React.DetailedHTMLProps<
      React.AnchorHTMLAttributes<HTMLAnchorElement>,
      HTMLAnchorElement
    >,
    "ref"
  > & {
    external?: boolean;
    newTab?: boolean;
  }
```

#### `<Button />` Props

```tsx
type ButtonProps = ButtonLinkProps &
  React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  >
```

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

> **Note**
> Not sure about some of the link diffs. This makes all links three things (bold, underline, primary) by default that is causing a large diff right now. Designers will be consulted before merging.

1. Check chromatic diff (it is large)
2. Run storybook and confirm button and links match [the mockup](https://www.figma.com/file/T1R64oNWog10OXiBnuiMkz/2.-Components?type=design&node-id=756-5888&t=MTBS0BmLaff44SI0-0) 

### Solid
Screenshot 2023-06-06 141639

### Inline
Screenshot 2023-06-06 141646

### Call to Action (cta)
Screenshot 2023-06-06 141653

### CODE REVIEW

**Comment by vd1992** [comment]
😱 

image

File: apps/web/src/components/Pagination/Pagination.stories.tsx
Comment (petertgiles) [inline apps/web/src/components/Pagination/Pagination.stories.tsx:1]: Not possible to see active page.
image

File: apps/web/src/components/UserProfile/UserProfile.tsx
Comment (petertgiles) [inline apps/web/src/components/UserProfile/UserProfile.tsx:1]: Was this change intentional?
image

File: apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx
Comment (petertgiles) [inline apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx:1]: These don't really look disabled anymore.
image

Comment (petertgiles) [reply apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx:1]: And this button also got super-sized.
image

Comment (petertgiles) [reply apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx:1]: We've also lost the indication of which step is active.
image

File: apps/web/src/pages/Home/HomePage/components/Profile/Profile.tsx [hunk 1]
```diff
@@ -77,17 +77,18 @@ const Profile = () => {
             data-h2-gap="base(x1)"
             data-h2-justify-content="base(center) p-tablet(flex-start)"
           >
-            <CallToActionLink
+            <Link
               color="quinary"
+              mode="cta"
               href={paths.myProfile()}
-              Icon={ProfileIcon}
+              icon={UserGroupIcon}
```
Comment (petertgiles) [inline apps/web/src/pages/Home/HomePage/components/Profile/Profile.tsx:84]: Intentional change?
image

File: apps/web/src/pages/Home/HomePage/components/Opportunities/Opportunities.tsx
Comment (petertgiles) [inline apps/web/src/pages/Home/HomePage/components/Opportunities/Opportunities.tsx:1]: Why is one link strong and the other is not?
image

File: apps/web/src/pages/Home/IAPHomePage/components/LanguageSelector.tsx
Comment (petertgiles) [inline apps/web/src/pages/Home/IAPHomePage/components/LanguageSelector.tsx:1]: Broken in mobile view.
image

Comment (petertgiles) [reply apps/web/src/pages/Home/IAPHomePage/components/LanguageSelector.tsx:1]: And non-mobile, too.  :laughing: 
image

**Review by petertgiles — CHANGES_REQUESTED** [review]
Very cool!  I haven't checked the code yet but here are the results of my Chromatic review.

File: apps/web/src/components/Pagination/Pagination.stories.tsx
Comment (esizer) [reply apps/web/src/components/Pagination/Pagination.stories.tsx:1]: Hmm, this looks like a result of dropping the outline style. 

@Jerryescandon or @substrae Do you have a suggestion on how we can highlight the current page now that we do not have the outline style?

File: apps/web/src/pages/DirectivePage/DirectivePage.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/DirectivePage/DirectivePage.tsx:1]: I'm not sure why but this raises a good question. Right now our links are using primary by default. @substrae Should inline links always be black by default?

File: apps/web/src/components/Pagination/Pagination.stories.tsx
Comment (Jerryescandon) [reply apps/web/src/components/Pagination/Pagination.stories.tsx:1]: My instinct is to change the #black for #primary-dark colour as highlight

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx
[the remark this answers is not in the document]
Comment (Jerryescandon) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:1]: ahmmm... let's keep them all strong + black for now. Im assuming at some point in the future, a big theme-ing issue will be made to update everything admin to the new look and feel. We can look at colours then.

File: apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx
Comment (vd1992) [reply apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx:1]: yeah, active step no longer shows but you can tell one is no longer clickable only when you hover over it

image

File: apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx
Comment (vd1992) [inline apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Some headers shifted, they weren't perfect to begin with though 

image

File: apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx
Comment (vd1992) [inline apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx:1]: This changed for buttons with lengthy text, is it intended?

image

File: apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx
[the remark this answers is not in the document]
Comment (vd1992) [reply apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx:1]: Appears to be the case only for single or so word 

image

File: apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx
Comment (petertgiles) [reply apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx:1]: Changes to padding, probably.  I just noticed in your screenshot that alignment changed, as well.

File: apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx
Comment (vd1992) [reply apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Oh, and bolding changed

File: apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx
[the remark this answers is not in the document]
Comment (Jerryescandon) [reply apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx:1]: ship it

[the remark this answers is not in the document]
Comment (Jerryescandon) [reply apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx:1]: image

[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx:1]: I mean, that looks a _little_ nicer than my mockup 😛 I can convert this into a new ticket and then just work on the alignment a bit for this PR.

Thanks @Jerryescandon 😄

File: apps/web/src/components/Pagination/Pagination.stories.tsx
Comment (esizer) [reply apps/web/src/components/Pagination/Pagination.stories.tsx:1]: Thanks! Changed made in [fix current page colour](https://github.com/GCTC-NTGC/gc-digital-talent/commit/32e29c82b375017188262bd17540c31d4f7c949e)

File: apps/web/src/components/UserProfile/UserProfile.tsx
Comment (esizer) [reply apps/web/src/components/UserProfile/UserProfile.tsx:1]: Odd, those don't even use any of the components I changed 😕 Good catch either way, fixed in [fix tab trigger styles](https://github.com/GCTC-NTGC/gc-digital-talent/commit/3fdecab793ccda7da7e5b2757cf6f98b8be28ee6)

File: apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx
Comment (esizer) [reply apps/web/src/pages/Applications/ApplicationReviewPage/ApplicationReviewPage.tsx:1]: Oops, I forgot we had this one disabled link 😆 Fixed with [fix disabled link styles](https://github.com/GCTC-NTGC/gc-digital-talent/commit/6d2f8d82af42519c7603b6afa8a77e5b3de8e693)

File: apps/web/src/pages/Home/HomePage/components/Profile/Profile.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/pages/Home/HomePage/components/Profile/Profile.tsx:84]: Not intentional 😬 Reverted in [fix incorrect profile icon](https://github.com/GCTC-NTGC/gc-digital-talent/commit/69bb1914b451643370da3b12b817252a5305aba3)

File: apps/web/src/pages/Home/IAPHomePage/components/LanguageSelector.tsx
Comment (esizer) [reply apps/web/src/pages/Home/IAPHomePage/components/LanguageSelector.tsx:1]: Okay, we can now pass `utilityIcon` similar to how we pass `icon`. This helps immensely with alignment and consistency. Updated in [fix icons after text alignment](https://github.com/GCTC-NTGC/gc-digital-talent/commit/7106dbfad2b679462b020535a8e62663f181a597)

File: apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx
Comment (esizer) [reply apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx:1]: So, with some of the alignment going on now for icons and the new call to action mode, having HTML in links causes some problems. We don't do this really ever except here so I just added a `span`. Not sure if it is worth the extra markup to do this for all links by default or not 🤔 


Anyway, fixed in [fix apply buttons](https://github.com/GCTC-NTGC/gc-digital-talent/commit/b6c8615b263d66d2084f44e5534ff9f1a894d630)

File: apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/EditPoolPage/EditPoolPage.tsx:1]: Alright, I at least for the alignment on the tall buttons done until we can update the layout.

[fix tall button text alignment](https://github.com/GCTC-NTGC/gc-digital-talent/commit/d4eb55e01504d4afe911991cf9af320de0f8bed1)

File: apps/web/src/pages/DirectivePage/DirectivePage.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/DirectivePage/DirectivePage.tsx:1]: Alright, after speaking with both @substrae and @BrookeCameron there has been some tweaks made which resulted in some large commit diffs 😅 The following decisions were made:

- Links are `secondary` by default to be consistent and stand out more compared to paragraph text
- New default mode for links `"text"`. This is essentially `"inline"` but lighter weight and styled to match paragraph text a little better

Based on this, I did a full review of all usage of `a` and `Link` and updated to all use `Link`. I then attempted to adjust the props of each link to conform to the decisions made around links. I did that across two fairly large commits. Hopefully I caught them all and this makes sense 😅 

- [replace anchors with Link component](https://github.com/GCTC-NTGC/gc-digital-talent/commit/0c712a6f27ac77ec63cd3efb7186e4185244c7da)
- [fix link weight, colour and alignment](https://github.com/GCTC-NTGC/gc-digital-talent/commit/8359533644a5c4ff57833f984bbde6596fd0fc3f)

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:1]: See: https://github.com/GCTC-NTGC/gc-digital-talent/pull/6813#discussion_r1222016513

File: apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx
Comment (vd1992) [inline apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Contrast issue with the pagination stuff at the bottom now, not sure if there are tables with a different colour theme present to check too

image

Comment (vd1992) [reply apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Sorry I don't quite follow, the comment you linked has to do with links and not table columns as far as I can tell.

Comment (esizer) [reply apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Those are technically the same thing just a different underlying HTML component. It was part of the [fix link weight, colour and alignment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6813/commits/8359533644a5c4ff57833f984bbde6596fd0fc3f) commit

Comment (petertgiles) [reply apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Yeah, Axe is flagging this.
image

Comment (esizer) [reply apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Oof, I did not realize we were using the button colour to also style the text 😬 Should be an easy fix, good catch 👍

Comment (esizer) [reply apps/web/src/components/Table/ApiManagedTable/BasicTable.tsx:1]: Alright, added a prop so we can control font colour separately from the button colour and updated both instances to use white text.

[fix pagination font colour](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6813/commits/76acecacb709c8f6931aa168cee406182c12217a)

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:1]: heh, hopefully I got it right with this one. I just added a `span` to all links regardless of content. Tried to locate all links that had `abbr` in them and confirm it was fixed. Guess the diff will surface it if I missed any 🤷‍♀️ 

[fix no icon link alignment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6813/commits/dc5d9c83c057d622046663c796108ae808d28558)

File: apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx
Comment (petertgiles) [reply apps/web/src/pages/Pools/BrowsePoolsPage/BrowsePoolsPage.tsx:1]: Hmm, still happening for me in Chromatic.
image

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx
[the remark this answers is not in the document]
Comment (petertgiles) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:1]: Hmm, something about this fix is making Axe quite upset.
image

[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:1]: That is actually a known problem but was a decision we made to make acronyms better for AT users. We really shouldn't be using acronyms as often as we do but it is difficult since we sort of need to in places like this. While we shouldn't be doing this, it is a necessary evil 😢

File: packages/ui/src/components/Alert/Alert.stories.tsx [hunk 2]
```diff
@@ -40,8 +41,10 @@ const TemplateAlert: ComponentStory<typeof Alert.Root> = ({
             {type === "error" && (
               <Alert.Footer>
                 <p>
-                  <a href="/#">Reach out to our support team</a> if you have any
-                  questions.
+                  <Link external href="/#">
+                    Reach out to our support team
+                  </Link>{" "}
```
Comment (petertgiles) [inline packages/ui/src/components/Alert/Alert.stories.tsx:46]: Stray space

**Review by petertgiles — CHANGES_REQUESTED** [review]
Code looks super clean!  Love the consolidation of ExternalLink CallToActionLink, IconLink.  Love the little tidy-ups.  I've noted a few things to check on below.

File: apps/web/src/pages/Auth/RegisterPage/RegisterPage.tsx [hunk 3]
```diff
@@ -13,7 +13,9 @@ import SEO from "~/components/SEO/SEO";
 import Hero from "~/components/Hero/Hero";
 
 const keyRegistrationLink = (path: string, chunks: React.ReactNode) => (
-  <a href={path}>{chunks}</a>
+  <Link external href={path}>
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Auth/RegisterPage/RegisterPage.tsx:16]: 🤔 Yeah, that is actually not a bad idea! I have never written an eslint rule, do you have an example or resource to point me towards? 😃

[the remark this answers is not in the document]
Comment (petertgiles) [reply apps/web/src/pages/Auth/RegisterPage/RegisterPage.tsx:16]: The last one I added was the restricted imports, here:
https://github.com/GCTC-NTGC/gc-digital-talent/blob/963595ff8180f09027fe6b78f9e159c54c7cd4eb/packages/eslint-config-custom/index.js#L135

It looks like this rule might do the trick:
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/forbid-elements.md

File: apps/web/src/pages/Applications/ApplicantDashboardPage/components/QualifiedRecruitments/QualifiedRecruitmentStatus.tsx [hunk 4]
```diff
@@ -85,7 +85,7 @@ const QualifiedRecruitmentStatus = ({
             id: "TDikDt",
             description:
               "Not suspended recruitment status message on qualified recruitment card.",
-          })}
+          })}{" "}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Applications/ApplicantDashboardPage/components/QualifiedRecruitments/QualifiedRecruitmentStatus.tsx:88]: These were actually intentional, the buttons to open the status dialog were bumping right up against the text next to them and I considered that part of the cleanup for buttons and links.

File: apps/web/src/pages/Applications/ApplicantDashboardPage/components/QualifiedRecruitments/QualifiedRecruitmentStatus.tsx [hunk 5]
```diff
@@ -157,7 +157,7 @@ const QualifiedRecruitmentStatus = ({
           id: "bqWmTD",
           description:
             "Suspended recruitment status message on qualified recruitment card.",
-        })}
+        })}{" "}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Applications/ApplicantDashboardPage/components/QualifiedRecruitments/QualifiedRecruitmentStatus.tsx:160]: Same thing here, this was intentional 😃 (what a creepy smile)

File: packages/ui/src/components/Alert/Alert.stories.tsx (hunk 2 above)
Comment (esizer) [reply packages/ui/src/components/Alert/Alert.stories.tsx:46]: Intentional, spaces from the text after.

File: apps/web/src/pages/Errors/ErrorPage/ErrorPage.tsx [hunk 6]
```diff
@@ -84,28 +84,30 @@ const ErrorPage = () => {
...
                 defaultMessage: "Go to the homepage",
                 id: "i9E0ka",
                 description: "Link text to go to the homepage from a 404",
               })}
-            </CallToActionLink>
-            <CallToActionLink
-              Icon={SupportIcon}
+            </Link>
+            <Link
+              icon={MagnifyingGlassIcon}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Errors/ErrorPage/ErrorPage.tsx]: Intentional, though I noticed a copy paste mistake with the name of the import 😬 Fixed in [Fix icon import name](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6813/commits/6685de96d1346a16be2896564a7694cc968ad01e)

File: apps/web/src/pages/Home/HomePage/components/Hero/Hero.tsx [hunk 7]
```diff
@@ -89,28 +89,30 @@ const Hero = ({ defaultImage }: HeroProps) => {
           data-h2-justify-content="base(center) p-tablet(flex-start)"
           data-h2-flex-wrap="base(wrap) p-tablet(initial)"
         >
-          <CallToActionLink
+          <Link
             color="quaternary"
-            Icon={JobIcon}
+            mode="cta"
+            icon={MagnifyingGlassIcon}
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Home/HomePage/components/Hero/Hero.tsx:95]: So, the icon names changed since they were custom and have been replaced by the heroicon equivalents. You should notice that they are the exact same icon, just imported from `@heroicons/react` with a different name.

I checked and confirmed this for all other icons you noted.

**Review by petertgiles — APPROVED** [review]
Very cool - I'm satisfied.  This one was yuge!

I'm thinking that after Vachan's comments are resolved, it might be worth asking Josh to scroll through Chromatic for a final check.

**Comment by substrae** [comment]
@esizer @petertgiles 

The Chromatic diff looks good to me, gents! Looks like you have it well in hand :)
